dynamiq/nodes/tools/mcp_adapter.py

```diff
@@ -0,0 +1,258 @@
+        Executes the MCP tool synchronously with the provided input.
+
+        Args:
+            input_data (BaseModel): Input data for the tool execution.
+            config (RunnableConfig, optional): Configuration for the runnable, including callbacks.
+            **kwargs: Additional arguments passed to the execution context.
+
+        Returns:
+            dict[str, Any]: A dictionary containing the tool's output.
+        """
+        return asyncio.run(self.execute_async(input_data, config, **kwargs))
+
+    async def execute_async(
+        self, input_data: BaseModel, config: RunnableConfig | None = None, **kwargs
+    ) -> dict[str, Any]:
+        """
+        Executes the MCP tool asynchronously.
+
+        Args:
+            input_data (BaseModel): Input data for the tool execution.
+
+        Returns:
+            dict[str, Any]: A dictionary containing the tool's output.
+        """
+        logger.info(f"Tool {self.name} - {self.id}: started with input:\n{input_data.model_dump()}")
+        config = ensure_config(config)
+        self.run_on_node_execute_run(config.callbacks, **kwargs)
+
+        input_dict = input_data.model_dump()
+
+        try:
+            async with await self.connection.get_client() as (read, write):
+                async with ClientSession(read, write) as session:
+                    await session.initialize()
+                    result = await session.call_tool(self.name, input_dict)
+        except Exception as e:
+            logger.error(f"Tool {self.name} - {self.id}: failed to get results. Error: {str(e)}")
+            raise ToolExecutionException(
+                f"Tool '{self.name}' failed to call tool from the MCP server."
+                f"Error: {str(e)}. Please analyze the error and take appropriate action.",
+                recoverable=True,
+            )
+
+        logger.info(f"Tool {self.name} - {self.id}: finished with result:\n{str(result)[:200]}...")
+        return {"content": result}
+
+
+class MCPServerAdapter(ConnectionNode):
+    """
+    A tool that manages connections to MCP servers and initializes MCP tools.
+
+    Attributes:
+      group (Literal[NodeGroup.TOOLS]): Node group.
+      name (str): Node name.
+      description (str): Node description.
+      connection (MPCConnection): The connection module with parameters needed to the MCP server.
+      tool_filter_names (list[str]): Names of tools to include or exclude.
+      selection_mode (ToolSelectionMode): Strategy for tool filtering (SELECT or DESELECT).
+      _mcp_tools (dict[str, MCPTool]): Internal dict of initialized MCP tools.
+    """
+
+    group: Literal[NodeGroup.TOOLS] = NodeGroup.TOOLS
+    name: str = "MCP Adapter Tool"
+    description: str = "The tool used to initialize available MCP tools based on provided server parameters."
+    connection: MPCConnection
+
+    tool_filter_names: list[str] = field(default_factory=list)
+    selection_mode: ToolSelectionMode = ToolSelectionMode.SELECT
```

Comment:
let's gather both with the same prefix and maybe use `include/exclude` as we used such prefixes for other tools params
```suggestion
    tool_filter_mode: ToolSelectionMode = ToolSelectionMode.SELECT
```